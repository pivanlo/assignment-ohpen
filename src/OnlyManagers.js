import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import { Link, Redirect } from "react-router-dom";
import { Auth } from "aws-amplify";

class OnlyManagers extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      username: props.auth.user && props.auth.user.username,
      email: props.auth.user && props.auth.user.attributes.email
    };

    this.handleSignOutClick = this.handleSignOutClick.bind(this);
  }

  async handleSignOutClick(event) {
    event.preventDefault();

    try {
      await Auth.signOut();

      this.props.auth.setAuthStatus(false);
      this.props.auth.setUser(null);
    } catch(error) {
      console.log(error);
    }
  }

  render() {
    if (!this.props.auth.isManager()) {
      return <Redirect to='/' />;
    }

    return (
      <div>
        <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
          <Navbar.Brand href="#home">Ohpen</Navbar.Brand>
          <Nav className="ml-auto">
            <Nav.Link href="#" onClick={this.handleSignOutClick}>Sign out</Nav.Link>
          </Nav>
        </Navbar>

        <div className="container mt-5">
          <h1 className="mb-4">Only Managers</h1>
          <p>This page is only accesible to managers.</p>
        </div>
      </div>
    );
  }
}

export default OnlyManagers;
