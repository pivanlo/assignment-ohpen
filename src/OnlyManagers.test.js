import React from 'react';
import OnlyManagers from "./OnlyManagers.js";
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly for managers', () => {
  const auth = {
    user: {
      attributes: {
        email: 'john.doe@gmail.com'
      }
    },
    isAuthenticated: true,
    isManager: () => true
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <OnlyManagers auth={auth} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly for non-managers', () => {
  const auth = {
    user: {
      attributes: {
        email: 'john.doe@gmail.com'
      }
    },
    isAuthenticated: true,
    isManager: () => false
  };

  const tree = renderer
    .create(
      <BrowserRouter>
        <OnlyManagers auth={auth} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
