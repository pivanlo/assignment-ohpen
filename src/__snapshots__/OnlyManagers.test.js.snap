// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for managers 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Ohpen
    </a>
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="#"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Sign out
      </a>
    </div>
  </nav>
  <div
    className="container mt-5"
  >
    <h1
      className="mb-4"
    >
      Only Managers
    </h1>
    <p>
      This page is only accesible to managers.
    </p>
  </div>
</div>
`;

exports[`renders correctly for non-managers 1`] = `null`;
